contracts/pods/PodAdminGateway.sol

```diff
@@ -0,0 +1,219 @@
+/// @notice Acts as a gateway for admin functionality and vetos in the TRIBE governance pod system
+/// @dev Contract is intended to be set as the podAdmin for all deployed Orca pods. It then allows
+///
+contract PodAdminGateway is CoreRef, IPodAdminGateway {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    /// @notice Orca membership token for the pods. Handles permissioning pod members
+    IMemberToken private immutable memberToken;
+
+    /// @notice Pod factory which creates optimistic pods and acts as a source of information
+    IPodFactory private immutable podFactory;
+
+    constructor(
+        address _core,
+        address _memberToken,
+        address _podFactory
+    ) CoreRef(_core) {
+        require(_memberToken != address(0x0), "ZERO_ADDRESS");
+        require(_podFactory != address(0x0), "ZERO_ADDRESS");
+
+        memberToken = IMemberToken(_memberToken);
+        podFactory = IPodFactory(_podFactory);
+    }
+
+    ////////////////////////   GETTERS   ////////////////////////////////
+
+    /// @notice Calculate the specific pod admin role related to adding pod members
+    function getPodAddMemberRole(uint256 _podId) public pure returns (bytes32) {
+        return keccak256(abi.encode(_podId, "ORCA_POD", "POD_ADD_MEMBER_ROLE"));
+    }
+
+    /// @notice Calculate the pod admin role related to removing pod members
+    function getPodRemoveMemberRole(uint256 _podId)
+        public
+        pure
+        returns (bytes32)
+    {
+        return
+            keccak256(abi.encode(_podId, "ORCA_POD", "POD_REMOVE_MEMBER_ROLE"));
+    }
+
+    /// @notice Calculate the specific pod veto role, which allows an
+    function getPodVetoRole(uint256 _podId) public pure returns (bytes32) {
+        return keccak256(abi.encode(_podId, "ORCA_POD", "POD_VETO_ROLE"));
+    }
+
+    /// @notice Calculate the specific pod transfer admin role
+    function getPodTransferAdminRole(uint256 _podId)
+        public
+        pure
+        returns (bytes32)
+    {
+        return
+            keccak256(
+                abi.encode(_podId, "ORCA_POD", "POD_TRANSFER_ADMIN_ROLE")
+            );
+    }
+
+    /////////////////////////    ADMIN PRIVILEDGES       ////////////////////////////
+
+    /// @notice Transfer the pod admin address for a pod to another address
+    function transferPodAdmin(uint256 _podId, address newPodAdmin)
+        external
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.POD_ADMIN,
+            getPodTransferAdminRole(_podId)
+        )
+    {
+        require(newPodAdmin != address(0x0), "ZERO_ADDRESS");
+        _transferPodAdmin(_podId, newPodAdmin);
+    }
+
+    /// @notice Batch transfer the pod admin address for several pods
+    /// @dev Mass transfer of podAdmins only expected to be performed by GOVERNOR or POD_ADMIN
+    function batchTransferPodAdmins(
+        uint256[] memory _podIds,
+        address[] memory newPodAdmins
```

Comment:
gas opti: these should be able to to be calldata instead of memory because the function is external

Reply:
Good point